test/integration/scheduler/plugins/plugins_test.go

```diff
@@ -1585,6 +1585,7 @@ func TestUnReservePreBindPlugins(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SchedulerAsyncAPICalls, false)
```

Comment:
/cc @macsko 
/hold

We hit the failure in this `TestUnReservePreBindPlugins` test.
>     runtime.go:221: E0729 04:12:31.656303] UnhandledError: Error updating pod err="failed to merge API calls: unexpected error: call of type *apidispatcher.queuedAPICall is not of type *PodStatusPatchCall"
https://storage.googleapis.com/kubernetes-ci-logs/pr-logs/pull/133021/pull-kubernetes-integration/1950035892986974208/build-log.txt

And, the test failure is gone after I disabled the async APi call feature. 
So, this is either my NNN change does something wrong and triggers the error in the async api call, or the async api call side has some bug that this PR triggers.
Either way, we cannot release both features until we figure out the cause of the integration test failure.